admin/src/main/resources/io/buoyant/admin/js/spec/RoutersSpec.js

```diff
@@ -0,0 +1,62 @@
+"use strict";
+
+define([
+  'jQuery',
+  'lodash',
+  'spec/fixtures/metrics',
+  'src/routers'
+], function($, _, metricsJson, Routers) {
+  describe("Routers", function() {
+    var fakeMetricsCollector = function() {};
+    var routers = Routers(metricsJson, fakeMetricsCollector);
+
+    describe("clients", function() {
+      it("provides the clients associated with a specified router", function() {
+        var multiplierClients = routers.clients("multiplier");
+        var expectedLabels = ["$/inet/127.1/9092", "$/inet/127.1/9030", "$/inet/127.1/9029"];
+
+        expect(multiplierClients.length).toBe(3);
+        expect(_.map(multiplierClients, 'label')).toEqual(expectedLabels);
+
+        _.each(multiplierClients, function(client) {
+          expect(client.router).toBe("multiplier");
+          expect(client.color).toBeTruthy(); // nonempty string
+          expect(client.prefix.indexOf("rt/multiplier/dst/id")).toBe(0);
+        });
+      });
+
+      it("returns all clients if router is invalid or unspecified", function() {
+        expect(routers.clients("foobar").length).toEqual(10);
+        expect(routers.clients().length).toEqual(10);
```

Comment:
Good idea... I've added a simple toHaveLength matcher in this file, but may look at just including something (e.g. https://github.com/just-boris/jasmine-collection-matchers/blob/master/src/toHaveLength.js) instead. (Or write more of our own matchers and also have specs for these matchers).